velox/dwio/dwrf/test/ReaderTest.cpp

```diff
@@ -1737,6 +1737,40 @@ TEST(TestReader, fileColumnNamesReadAsLowerCaseComplexStruct) {
   EXPECT_EQ(col0_1_1_0_0->childByName("ccint3"), col0_1_1_0_0_0);
 }
 
+TEST(TestReader, testOrcCharType) {
+  // test_char.orc has schema: struct<d_date_id:char(16),type_id:char(1)>
+  auto pool = getDefaultPool().get();
+  dwio::common::ReaderOptions readerOpts{pool};
+  readerOpts.setFileFormat(dwio::common::FileFormat::ORC);
+
+  const std::shared_ptr<const RowType> expectedType =
+      std::dynamic_pointer_cast<const RowType>(HiveTypeParser().parse(
+          "struct<d_date_id:varchar(16),type_id:varchar(1)>"));
+
+  auto reader = DwrfReader::create(
+      createFileBufferedInput(
+          getExampleFilePath("test_char.orc"), readerOpts.getMemoryPool()),
+      readerOpts);
+  auto rowType = reader->rowType();
+  EXPECT_TRUE(rowType->equivalent(*expectedType));
+
+  RowReaderOptions rowReaderOpts;
+  auto rowReader = reader->createRowReader(rowReaderOpts);
+  auto actual = BaseVector::create(rowType, 0, pool);
+  ASSERT_EQ(rowReader->next(1024, actual), 2);
+  EXPECT_EQ(actual->size(), 2);
+
+  VectorMaker maker(pool);
```

Comment:
Can you make pool and maker member variables instead of creating them in every test? Alternatively, consider using VectorTestBase as the base class.